src/main/openapi/ds-image-analysis-openapi_v1.yaml

```diff
@@ -16,40 +31,110 @@ servers:
 
 paths:
   # These endpoints has been created from the colorizer endpoint-template
-  /maincolor:
+  /main-rgb-color:
     post:
       tags:
-        - 'Image Analysis'
-      summary: 'Get the most dominant color from an image.'
-      operationId: getMainColor
-
+        - 'Color Analysis'
+      summary: 'Get the most dominant colors from an image. Calculated in RGB space with euclidian distance.'
+      operationId: getMainRgbColor
       requestBody:
         content:
           multipart/form-data:
             schema:
               type: object
               required:
                 - image
+                - top-colors
               properties:
                 # The POSTed image can be retrieved using imageDetail.getDataHandler().getInputStream()
                 image:
                   description: 'The image to analyse'
                   type: string
                   format: binary
+                top-colors:
+                  description: 'Number of colors to return'
+                  type: integer
+                  format: int32
+                  default: 10
+      responses:
+        '200':
+          description: 'The RGB hex value of colors and percentage of image with given color.'
+          content:
+            application/json:
+              schema:
+                type: array
+                items: 
+                  type: object
+                  properties:
+                    hexRGB:
+                      type: string
+                    percent:
+                      type: number
+                      format: float
+                example:
+                - hexRGB: '#000000'
+                  percent: 8.38468
+                - hexRGB: '#FFFFFF'
+                  percent: 5.23782
 
+  /main-OKlab-colors:
```

Comment:
This is really heavy on non-trivial images, but we knew it would be so, so this it just a reminder that we need to do some optimization before we can use this in production.